balancer-js/src/modules/data/gyro-config/repository.ts

```diff
@@ -0,0 +1,63 @@
+import { formatBytes32String } from '@ethersproject/strings';
+import { keccak256 } from '@ethersproject/solidity';
+import { defaultAbiCoder } from '@ethersproject/abi';
+import { formatFixed } from '@ethersproject/bignumber';
+import { Provider } from '@ethersproject/providers';
+import { GyroConfig, GyroConfig__factory } from '@/contracts';
+
+export interface GyroConfigRepository {
+  getGyroProtocolFee(poolAddress: string): Promise<number>;
+}
+
+export class GyroConfigRepositoryImpl implements GyroConfigRepository {
+  gyroConfig: GyroConfig;
+
+  constructor(private gyroConfigAddress: string, provider: Provider) {
+    this.gyroConfig = GyroConfig__factory.connect(gyroConfigAddress, provider);
+  }
+
+  async getGyroProtocolFee(poolAddress: string): Promise<number> {
+    let fee = 0;
+    const protocolFeePercKey = formatBytes32String('PROTOCOL_SWAP_FEE_PERC');
+
+    const gyroPoolTypeKey = formatBytes32String('E-CLP');
+    const encodedPoolSpecificKey = keccak256(
+      ['bytes'],
+      [
+        defaultAbiCoder.encode(
+          ['bytes32', 'uint256'],
+          [protocolFeePercKey, poolAddress]
+        ),
+      ]
+    );
+
+    const encodedPoolTypeKey = keccak256(
+      ['bytes'],
+      [
+        defaultAbiCoder.encode(
+          ['bytes32', 'bytes32'],
+          [protocolFeePercKey, gyroPoolTypeKey]
+        ),
+      ]
+    );
+    const hasPoolSpecificKey = await this.gyroConfig.hasKey(
```

Comment:
To avoid multiple separate RPC calls we could batch these into a multicall similar to https://github.com/balancer/balancer-sdk/blob/9ae39b45f609a7b049c496d1e7cf9c7a380f7412/balancer-js/src/modules/data/fee-distributor/repository.ts#L83